lib/matplotlib/tests/test_build_grid.py

```diff
@@ -0,0 +1,56 @@
+from matplotlib.testing.decorators import check_figures_equal
```

Comment:
I think we usually put tests in the submodule that is being tested ie. `test_figure.py`?  In which case you'll need longer names below.  Slightly prefer `nested` to `recursive`.

Reply:
I think we should start splitting the tests up more.  It makes "run this file" for the tests more manageable and it reduces the annoying conflicts when two PRs add tests to the same file.

Reply:
IMHO there is value in having all tests of module `foo.py` in `test_foo.py` (or at least `test_foo_something.py` if you really need more separation). I don‘t think we have too many PR conflicts over tests. As for running tests 
`pytest path_to/test_foo.py`, `pytest path_to/test_foo.py::test_my_func` should work well enough.

Reply:
Could also start using testclasses to collect all the functions that test the same function?

Reply:
Fwiw I also prefer keeping everything in a single file per module.